### PR TITLE
Dtos and controllers(movie, movie session, register) added

### PR DESCRIPTION
**Pull request by margarita-ignatieva**

### CODE REVIEW

File: src/main/java/com/cinema/shop/controllers/MovieSessionController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    private MovieSessionMapper movieSessionMapper;
+
+    @PostMapping
+    public void addMovieSessionDto(@RequestBody MovieSessionRequestDto requestDto) {
+        movieSessionService.add(movieSessionMapper.getMovieSession(requestDto));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAll(@RequestParam Long id, @RequestParam
+            @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate date) {
```
Comment (KovalAlexey) [inline src/main/java/com/cinema/shop/controllers/MovieSessionController.java:37]: Nice solution, take note

File: src/main/java/com/cinema/shop/mappers/UserMapper.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import com.cinema.shop.model.User;
+import com.cinema.shop.model.dto.UserRequestDto;
+import com.cinema.shop.model.dto.UserResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMapper {
+
+    public User getUser(UserRequestDto userRequestDto) {
```
Comment (KovalAlexey) [inline src/main/java/com/cinema/shop/mappers/UserMapper.java]: Where do you use these methods?

**Review by KovalAlexey — APPROVED** [review]

File: src/main/java/com/cinema/shop/mappers/UserMapper.java (hunk 2 above)
Comment (margarita-ignatieva) [reply src/main/java/com/cinema/shop/mappers/UserMapper.java]: just to make standard dto transition

File: src/main/java/com/cinema/shop/Main.java [hunk 3]
```diff
@@ -89,5 +90,7 @@ public static void main(String[] args) {
 
         orderService.completeOrder(shoppingCartService.getByUser(shinoda).getTickets(), shinoda);
         log.info("Trying to get user orders: " + orderService.getOrderHistory(shinoda));
+
+        log.info(movieSession2.getShowTime().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
```
Comment (Sofasmile) [inline src/main/java/com/cinema/shop/Main.java]: we can remove Main class

File: src/main/java/com/cinema/shop/controllers/AuthenticationController.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.cinema.shop.model.dto.UserRequestDto;
+import com.cinema.shop.service.AuthenticationService;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/register")
```
Comment (Sofasmile) [inline src/main/java/com/cinema/shop/controllers/AuthenticationController.java]: https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_33.html

File: src/main/java/com/cinema/shop/dao/CinemaHallDao.java [hunk 5]
```diff
@@ -7,4 +7,6 @@
     CinemaHall add(CinemaHall cinemaHall);
 
     List<CinemaHall> getAll();
+
+    CinemaHall findById(Long id);
```
Comment (Sofasmile) [inline src/main/java/com/cinema/shop/dao/CinemaHallDao.java]: what is the difference between  findById() and getById()?? let's use one type of naming

File: src/main/java/com/cinema/shop/mappers/UserMapper.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package com.cinema.shop.mappers;
+
+import com.cinema.shop.model.User;
+import com.cinema.shop.model.dto.UserRequestDto;
+import com.cinema.shop.model.dto.UserResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserMapper {
```
Comment (Sofasmile) [inline src/main/java/com/cinema/shop/mappers/UserMapper.java]: let's remove this class
we don't use it in this hw

File: src/main/java/com/cinema/shop/model/dto/UserResponseDto.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package com.cinema.shop.model.dto;
+
+import lombok.Data;
+
+@Data
+public class UserResponseDto {
```
Comment (Sofasmile) [inline src/main/java/com/cinema/shop/model/dto/UserResponseDto.java]: this is redundant now

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**